src/dom-manipulation.js

```diff
@@ -214,10 +234,17 @@ export const getNativeNode = (element) => {
   // While ReactDOM's `findDOMNode` is discouraged, it's the only
   // publicly-exposed way to find the underlying DOM node for
   // composite components.
-  return findDOMNode(element);
+  const foundNode: ?(Element | Text) = findDOMNode(element);
+
+  if (foundNode instanceof HTMLElement) {
+    return foundNode;
+  }
+
+  // Text nodes are not supported
+  return null;
```

Comment:
Good catch.

Nitpick: I find it clearer when invalid cases are dealt with before the expected case, so that the last line in a function is the desired result. I would reorder:

```js
const isHTMLElement = foundNode instanceof HTMLElement;

if (!isHTMLElement) {
  // Text nodes are not supported
  return null;
}

return foundNode;
```